sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3646,10 +3646,12 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
     )
   }
 
-  def implicitCollationMismatchError(): Throwable = {
+  def implicitCollationMismatchError(implicitTypes: Seq[AbstractDataType]): Throwable = {
```

Comment:
Why AbstractDataType? Shouldn't we always get StringTypes?

Reply:
sure, we can limit to StringType only here